compression.cpp: error: 'isnan' was not declared in this scope; did you mean 'std::isnan'?

```
:info:build cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/build/src/examples && /opt/local/bin/g++-mp-12  -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/build/src/examples -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/examples -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/lib/OpenEXR -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/build/cmake -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/lib/Iex -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/lib/IlmThread -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/lib/OpenEXRCore -isystem /opt/local/include -isystem /opt/local/include/Imath -pipe -Os -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/examples/CMakeFiles/OpenEXRExamples.dir/previewImageExamples.cpp.o -MF CMakeFiles/OpenEXRExamples.dir/previewImageExamples.cpp.o.d -o CMakeFiles/OpenEXRExamples.dir/previewImageExamples.cpp.o -c /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/examples/previewImageExamples.cpp
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/test/OpenEXRCoreTest/compression.cpp: In function 'bool withinDWAErrorBounds(uint16_t, uint16_t)':
:info:build /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/test/OpenEXRCoreTest/compression.cpp:162:10: error: 'isnan' was not declared in this scope; did you mean 'std::isnan'?
:info:build   162 |     if (!isnan (a1))
:info:build       |          ^~~~~
:info:build       |          std::isnan
:info:build In file included from /opt/local/include/LegacySupport/cmath:54,
:info:build                  from /opt/local/include/Imath/ImathMath.h:16,
:info:build                  from /opt/local/include/Imath/ImathVec.h:17,
:info:build                  from /opt/local/include/Imath/ImathBox.h:16,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/lib/OpenEXR/ImfCompressor.h:19,
:info:build                  from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_graphics_openexr/openexr/work/openexr-7c40603eb2812148c6a99fbb6ee07533395fdadb/src/test/OpenEXRCoreTest/compression.cpp:31:
:info:build /opt/local/include/gcc12/c++/cmath:632:5: note: 'std::isnan' declared here
:info:build   632 |     isnan(_Tp __x)
:info:build       |     ^~~~~
```

Unfortunately, this does not solve the problem:

```
:info:build [ 66%] Building CXX object src/test/OpenEXRTest/CMakeFiles/OpenEXRTest.dir/compareDwa.cpp.o
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/src/test/OpenEXRTest && /opt/local/bin/g++-mp-12 -DILM_IMF_TEST_IMAGEDIR=\"/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/test/OpenEXRTest/\" -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/src/test/OpenEXRTest -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/test/OpenEXRTest -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/OpenEXR -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/cmake -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/Iex -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/IlmThread -isystem /opt/local/include -isystem /opt/local/include/Imath -pipe -Os -D__STDC_FORMAT_MACROS -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/test/OpenEXRTest/CMakeFiles/OpenEXRTest.dir/compareDwa.cpp.o -MF CMakeFiles/OpenEXRTest.dir/compareDwa.cpp.o.d -o CMakeFiles/OpenEXRTest.dir/compareDwa.cpp.o -c /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/test/OpenEXRTest/compareDwa.cpp
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/test/OpenEXRCoreTest/compression.cpp: In function 'bool withinDWAErrorBounds(uint16_t, uint16_t)':
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/test/OpenEXRCoreTest/compression.cpp:162:10: error: 'isnan' was not declared in this scope; did you mean 'std::isnan'?
:info:build   162 |     if (!isnan (a1))
:info:build       |          ^~~~~
:info:build       |          std::isnan
:info:build In file included from /opt/local/include/LegacySupport/cmath:54,
:info:build                  from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/test/OpenEXRCoreTest/compression.cpp:26:
:info:build /opt/local/include/gcc12/c++/cmath:632:5: note: 'std::isnan' declared here
:info:build   632 |     isnan(_Tp __x)
:info:build       |     ^~~~~
:info:build [ 66%] Building CXX object src/examples/CMakeFiles/OpenEXRExamples.dir/main.cpp.o
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/src/examples && /opt/local/bin/g++-mp-12  -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/src/examples -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/examples -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/OpenEXR -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/cmake -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/Iex -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/IlmThread -isystem /opt/local/include -isystem /opt/local/include/Imath -pipe -Os -D__STDC_FORMAT_MACROS -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/examples/CMakeFiles/OpenEXRExamples.dir/main.cpp.o -MF CMakeFiles/OpenEXRExamples.dir/main.cpp.o.d -o CMakeFiles/OpenEXRExamples.dir/main.cpp.o -c /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/examples/main.cpp
:info:build [ 66%] Building CXX object src/bin/exrenvmap/CMakeFiles/exrenvmap.dir/readInputImage.cpp.o
:info:build cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/src/bin/exrenvmap && /opt/local/bin/g++-mp-12  -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/src/bin/exrenvmap -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/bin/exrenvmap -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/OpenEXR -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build/cmake -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/Iex -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/lib/IlmThread -isystem /opt/local/include -isystem /opt/local/include/Imath -pipe -Os -D__STDC_FORMAT_MACROS -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/bin/exrenvmap/CMakeFiles/exrenvmap.dir/readInputImage.cpp.o -MF CMakeFiles/exrenvmap.dir/readInputImage.cpp.o.d -o CMakeFiles/exrenvmap.dir/readInputImage.cpp.o -c /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/openexr-3.1.8/src/bin/exrenvmap/readInputImage.cpp
:info:build make[2]: *** [src/test/OpenEXRCoreTest/CMakeFiles/OpenEXRCoreTest.dir/compression.cpp.o] Error 1
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_graphics_openexr/openexr/work/build'
:info:build make[1]: *** [src/test/OpenEXRCoreTest/CMakeFiles/OpenEXRCoreTest.dir/all] Error 2
:info:build make[1]: *** Waiting for unfinished jobs....
```

--- a/src/test/OpenEXRCoreTest/compression.cpp
+++ b/src/test/OpenEXRCoreTest/compression.cpp
@@ -159,7 +159,7 @@ inline bool
 withinDWAErrorBounds (const uint16_t a, const uint16_t b)
 {
     float a1 = imath_half_to_float (a);
-    if (!isnan (a1))
+    if (!std::isnan (a1))
     {
         float a2 = imath_half_to_float (b);
         float denominator =

